php-checks/src/main/java/org/sonar/php/checks/PHPDeprecatedFunctionUsageCheck.java

```diff
@@ -166,27 +168,32 @@ public void visitFunctionDeclaration(FunctionDeclarationTree tree) {
 
   @Override
   protected void checkFunctionCall(FunctionCallTree tree) {
-    String functionName = ((NamespaceNameTree) tree.callee()).qualifiedName();
+    ExpressionTree callee = tree.callee();
+    // The object creations can be ignored. Only functions are deprecated.
+    if (callee instanceof ClassNamespaceNameTreeImpl) {
+      return;
+    }
+    String functionName = ((NamespaceNameTree) callee).qualifiedName();
 
     if (SET_LOCALE_FUNCTION.equalsIgnoreCase(functionName)) {
       checkLocalCategoryArgument(tree.callArguments());
-
     } else if (PARSE_STR_FUNCTION.equalsIgnoreCase(functionName)) {
       checkParseStrArguments(tree);
-
     } else if (ASSERT_FUNCTION.equalsIgnoreCase(functionName)) {
       checkAssertArguments(tree);
-
     } else if (DEFINE_FUNCTION.equalsIgnoreCase(functionName)) {
       checkDefineArguments(tree);
-
     } else if (SEARCHING_STRING_FUNCTIONS.contains(functionName.toLowerCase(Locale.ROOT))) {
       checkSearchingStringArguments(tree);
-
-    } else {
+    } else if (!ifFunctionRedefinedInCurrentNamespace(tree, functionName)) {
       context().newIssue(this, tree.callee(), buildMessage(functionName.toLowerCase(Locale.ROOT)));
     }
+  }
 
+  private static boolean ifFunctionRedefinedInCurrentNamespace(FunctionCallTree tree, String functionName) {
+    return tree.is(Kind.FUNCTION_CALL) &&
```

Comment:
As you have a FunctionCallTree as an argument, why do you check this? Isn't it redundant?